README.md

```diff
@@ -2,11 +2,110 @@
+     dependencies {
+         implementation 'com.palantir.conjure.java:conjure-lib:0.0.1'
+     }
+     """
+```
+
+or in some "Versions" class that has a listing of all the dependencies the test uses.  e.g. 
+```java
+final class TestPluginVersions {
+    static final String CONJURE_JAVA = "com.palantir.conjure.java:conjure-java:5.7.1";
+    static final String CONJURE = "com.palantir.conjure:conjure:4.10.1";
+```
+
+Once these tests are written, the versions of the plugins are often not updated, even when the project under test keeps its dependencies of those plugins up-to-date.  This can cause tests to fail when the plugin is updated not because the plugin is bad, but because there is an incompatibility in the old versions of plugins used in the integration tests.  This can often happen with gradle version bumps.
+
+When applied to a project, the plugin-testing plugin scans the `testRuntimeClasspath` configuration for the project and passes all dependencies to the test task as a system property.  The version of the dependencies can then be resolved when tests are run and written into generated files.  e.g.
+
+```groovy
+import static com.palantir.gradle.plugintesting.TestDepVersions.resolve
+import nebula.test.IntegrationSpec
+
+class HelloWorldSpec extends IntegrationSpec {
+   def setup() {
+       //language=gradle
+       buildFile << """
+        buildscript {
+            repositories {
+                mavenCentral()
+            }
+            dependencies {
+                classpath '${resolve('com.palantir.gradle.conjure:gradle-conjure')}'
+            }
+        }
+        dependencies {
+            implementation '${resolve('com.palantir.conjure.java:conjure-lib')}'
+        }
+        """
+   }
+```
+# Resolution of gradle versions to test against
+Similarly, tests may hardcode versions of gradle that they need to stay compatible with. These versions also get stale and PRs start failing for the inverse reason of the above - the code in the plugin or a dependency of it is updated and is no longer compatible with an old version of gradle. For example, attempting to update jackson libraries from 2.15 -> 2.17 would fail if a test tried to run on gradle < 7.6.4 (when compatibility with jackson 17 was fixed).
+
+The `GradleTestVersions` class can provide up-to-date versions of gradle to test against.  
+```groovy
+import nebula.test.IntegrationSpec
+import com.palantir.gradle.plugintesting.GradleTestVersions
+
+class HelloWorldSpec extends IntegrationSpec {
+    def 'runs on version of gradle: #version'() {
+        when:
+        gradleVersion = version
+
+        then:
+        def result = runTasks('someTask')
+        result.success
+
+        where:
+        version << GradleTestVersions.gradleVersionsForTests
+    }
+}
+```
+The plugin sets default versions to test against, but these can be overridden using the `gradleTestUtils` extension.  e.g.
+
+```groovy
+gradleTestUtils {
+    gradleVersions = ['7.6.4', '8.8']
+}
+```
```

Comment:
well, the reality is I don't think individual repos _should_ be overriding it because then you are essentially back to the original problem of each repo testing against old versions.  The larger solution is that an organization would write another plugin to set the standard versions that it wants and then apply that plugin to all repos.  This is similar to what we do with latest-jdks, dev-env configs, etc.  Or possibly we change the underlying component to look at a file in the repo with the versions and have an excavator go through and modify that file directly.

We can work out the best approach when we need to start moving versions from where we start them.  This just lays the groundwork for making it possible.